### PR TITLE
Issue 3275/fix remove used param

### PR DESCRIPTION
**Pull request by tclayton-newr**
## What's changed?
https://github.com/openrewrite/rewrite/issues/3275 pointed out an issue with incorrectly removing an import that was being used as a function parameter. This change updates the class path matching logic to check for either a `.` or a `$` class path separator. 

## What's your motivation?
My motivation is to resolve https://github.com/openrewrite/rewrite/issues/3275 and make the RemoveUnusedImports recipe more reliable for wider adoption. 

## Anything in particular you'd like reviewers to focus on?
I co-opted the predicate used in https://github.com/openrewrite/rewrite/pull/3363 to more generally match against class paths that have a `$` as a separator. 

## Anyone you would like to review specifically?
Any maintainer 😄 

## Have you considered any alternatives or workarounds?
n/a

## Any additional context
n/a

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've added the license header to any new files through `./gradlew licenseFormat`
- [x] I've used the IntelliJ auto-formatter on affected files

### CODE REVIEW

**Comment by knutwannheden** [comment]
@tclayton-newr Thanks for contributing this PR! I will take a look at it later today or tomorrow.

File: rewrite-java-test/src/test/java/org/openrewrite/java/RemoveUnusedImportsTest.java [hunk 1]
```diff
@@ -1072,6 +1072,68 @@ public static void main(String[] args) {
...
+
+              class Test {
+                  void f(A classA) {
+                    classA.getShort1();
+                  }
+              }
+              """
+          )
+        );
+
```
Comment (timtebeek) [inline rewrite-java-test/src/test/java/org/openrewrite/java/RemoveUnusedImportsTest.java:1106]: I'm wondering if this should be two tests; we don't usually see two invocations of `rewriteRun` in a single test method.

File: rewrite-java/src/main/java/org/openrewrite/java/RemoveUnusedImports.java [hunk 2]
```diff
@@ -285,6 +286,10 @@ public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu, ExecutionCon
         }
     }
 
+    private static Predicate<String> classPathMatcher(final String target){
+        return (s) -> s.matches(target.replaceAll("[\\.\\$]", "(\\\\\\.|\\\\\\$)"));
```
Comment (knutwannheden) [inline rewrite-java/src/main/java/org/openrewrite/java/RemoveUnusedImports.java:290]: I was going to suggest:
```suggestion
        return (s) -> s.matches(target.replaceAll("[.$]", "[.\\$]"));
```
but I now wonder if we can't simply do:
```suggestion
            return s -> TypeUtils.fullyQualifiedNamesAreEqual(s, target);
```

File: rewrite-java-test/src/test/java/org/openrewrite/java/RemoveUnusedImportsTest.java (hunk 1 above)
Comment (tclayton-newr) [reply rewrite-java-test/src/test/java/org/openrewrite/java/RemoveUnusedImportsTest.java:1106]: I am happy to split them out

File: rewrite-java/src/main/java/org/openrewrite/java/RemoveUnusedImports.java (hunk 2 above)
Comment (tclayton-newr) [reply rewrite-java/src/main/java/org/openrewrite/java/RemoveUnusedImports.java:290]: Interesting, I'll have a look at the second approach

**Review by joanvr — APPROVED** [review]

**Comment by joanvr** [comment]
Thank you very much for your contribution and addressing all the comments :)

**Pull request merged**